### PR TITLE
Add engineering manager distribution

### PR DESCRIPTION
**Pull request by nicksnyder**
This job description is somewhat sparse for a few reasons:
1. The kinds of experienced/savy EMs that we are looking for are going to get more value out of an intro call than a job description.
1. The Distribution team page should be the source of truth for a short/inspiring description of what the team is and why it exists.
1. Not sweating the details in this job description makes it easier to merge sooner. We already had someone cold email us even thought we didn't have a job description posted, and I want to signal to others looking for jobs that we are in fact looking.

### CODE REVIEW

**Comment by slimsag** [comment]
https://about.sourcegraph.com/handbook/engineering/distribution#ownership-areas

File: job-descriptions/engineering-manager-distribution.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+To create a product that serves the needs of all developers, we are building a diverse [all-remote team](https://about.sourcegraph.com/company/remote) that is [distributed across the world](https://about.sourcegraph.com/company/team).
+
+We provide [competitive compensation](https://about.sourcegraph.com/handbook/people-ops/compensation) and [practical benefits](https://about.sourcegraph.com/handbook/people-ops/benefits-and-perks) to keep you happy and healthy so that you can do your best work.
+
+Learn more about what it is like to work at Sourcegraph by reading [our handbook](https://about.sourcegraph.com/handbook/).
+
+## Interview process
+
+1. You [apply here](TODO).
```
Comment (beyang) [inline job-descriptions/engineering-manager-distribution.md:32]: Should we complete this link before merging?

File: job-descriptions/engineering-manager-distribution.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+1. We schedule a call with our VP of Engineering to learn more about you and answer any questions that you have about Sourcegraph.
+1. We schedule multiple remote interviews to determine if you would be a good fit for our team.
+1. We check your references.
+1. We make you a job offer.
+
+We want to ensure Sourcegraph is an environment that suits your working style and empowers you to do your best work, so we are eager to answer any questions that you have about us at any point in the interview process.
+
+**[Click here to apply](TODO)**
```
Comment (beyang) [inline job-descriptions/engineering-manager-distribution.md:40]: Finish link?

**Comment by nicksnyder** [comment]
Going to merge this and iterate on the description in a separate PR.

**Pull request merged**